gossip/client_test.go

```diff
@@ -138,26 +138,32 @@ func TestClientGossip(t *testing.T) {
 		}
 	}()
 
-	if err := local.AddInfo("local-key", nil, time.Second); err != nil {
+	if err := local.AddInfo("local-key", nil, time.Hour); err != nil {
 		t.Fatal(err)
 	}
-	if err := remote.AddInfo("remote-key", nil, time.Second); err != nil {
+	if err := remote.AddInfo("remote-key", nil, time.Hour); err != nil {
 		t.Fatal(err)
 	}
 
 	// Use an insecure context. We're talking to tcp socket which are not in the certs.
 	rpcContext := rpc.NewContext(&base.Context{Insecure: true}, nil, stopper)
-	client.start(local, disconnected, rpcContext, stopper)
+	disconnected <- client
 
-	util.SucceedsSoon(t, func() error {
```

Comment:
@tamird The test was also flaky because the gRPC connection failed to dial within its timeout (3 sec).